### ISSUE TITLE
Meaning of JobManagerInterface defaults

### ISSUE THREAD

**Issue by dkfellows** [SpiNNakerManchester/RemoteSpiNNaker#6]
All operations in the `JobManagerInterface` that have default values should specify what happens with those defaults. For example, with `getJobMachine` (mapped a `GET` on the path `/job/{id}/machine`), at most one of the `nCores`, `nChips` and `nBoards` query parameters should be set, and all can be omitted. However, for the `runTime` query parameter there's a default but it doesn't make sense.

What should happen if the expected(?) running time is unspecified? What does that mean?

**Comment by rowleya** [comment]
Yes, some comments here would have been useful!  It actually looks like this can't be -1, and so this is an error.

The idea of "-1" (or unspecified) run time was that it means "run forever".  However, that has the implication that the quota to be used is infinite, which will almost certainly fail!  That said, it is possible to not give a run time to a script and then stop the job later in the script with some criteria (e.g. receive live spikes and stop when you have enough spikes or are to have deemed to have learned some pattern).

Maybe this has to be "unsupported" on the portal at present though, as otherwise this becomes complicated; my suggestion is to remove the ability for unspecified run time for now and we can address the issue if a user requests this.
